crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -26,31 +27,42 @@ use crate::{
 
 /// If an entity is hidden in this way,  all [`Children`] (and all of their children and so on) will also be hidden.
 /// This is done by setting the values of their [`ComputedVisibility`] component.
-#[derive(Component, Clone, Reflect, Debug)]
+#[derive(Component, Clone, Copy, Reflect, Debug, PartialEq, Eq)]
 #[reflect(Component, Default)]
-pub struct Visibility {
-    /// Indicates whether this entity is visible. Hidden values will propagate down the entity hierarchy.
-    /// If this entity is hidden, all of its descendants will be hidden as well. See [`Children`] and [`Parent`] for
-    /// hierarchy info.
-    pub is_visible: bool,
+pub enum Visibility {
```

Comment:
Ensure the size of the type does not change via `repr(u8)`.

Reply:
From what I understand, the compiler should already use a `u8` for this enum, and in fact asking for it to unconditionally use a `u8` means the compiler won't be able to make [optimisations](https://github.com/rust-lang/rust/pull/45225) when you start nesting the type inside of other values like in an `Option` or `Result` (I could be wrong though, maybe the compiler is free to ignore the `repr(u8)` under certain circumstances?).

Reply:
If that is the case, please add a test to ensure that the size remains at 1 byte for both `Visibility` and `Option<Visibility>` . The performance of many rendering systems is reliant on this type staying small.